packages/core/src/integration.ts

```diff
@@ -54,31 +54,24 @@ export function getIntegrationsToSetup(options: Options): Integration[] {
   return integrations;
 }
 
-/** Setup given integration */
-export function setupIntegration(integration: Integration): void {
-  if (installedIntegrations.indexOf(integration.name) !== -1) {
-    return;
-  }
-  integration.setupOnce(addGlobalEventProcessor, getCurrentHub);
-  installedIntegrations.push(integration.name);
-  IS_DEBUG_BUILD && logger.log(`Integration installed: ${integration.name}`);
-}
-
 /**
  * Given a list of integration instances this installs them all. When `withDefaults` is set to `true` then all default
  * integrations are added unless they were already provided before.
  * @param integrations array of integration instances
  * @param withDefault should enable default integrations
  */
-export function setupIntegrations<O extends ClientOptions>(options: O): IntegrationIndex {
-  const integrations: IntegrationIndex = {};
-  options.integrations.forEach(integration => {
-    integrations[integration.name] = integration;
-    setupIntegration(integration);
+export function setupIntegrations(integrations: Integration[]): IntegrationIndex {
+  const integrationIndex: IntegrationIndex = {};
+
+  integrations.forEach(integration => {
+    integrationIndex[integration.name] = integration;
+
+    if (installedIntegrations.indexOf(integration.name) === -1) {
+      integration.setupOnce(addGlobalEventProcessor, getCurrentHub);
+      installedIntegrations.push(integration.name);
+      IS_DEBUG_BUILD && logger.log(`Integration installed: ${integration.name}`);
+    }
```

Comment:
More of a question than anything else, especially since this PR doesn't change anything in this regard : Can `setupOnce` fail? Currently, each integration is added to `integrationIndex` (which according to the JSdoc holds all set up integrations) before it is installed. This means that the index _could_ hold integrations that actually _aren't_ installed if `setupOnce` fails (?).